packages/http/src/mocker/__tests__/HttpMocker.spec.ts

```diff
@@ -229,6 +228,118 @@ describe('HttpMocker', () => {
+                          allOf: [{ type: 'string' }, { default: 'hello' }],
+                        },
+                        pet: {
+                          type: 'object',
+                          properties: {
+                            name: { type: 'string', examples: ['Clark'] },
+                            middlename: { type: 'string', examples: ['J'], default: 'JJ' },
+                          },
+                        },
+                      },
+                      required: ['name', 'surname', 'age', 'email'],
+                    },
+                    encodings: [],
+                  },
+                ],
+              },
+            ],
+          };
+
+          const response = httpMocker.mock({
+            input: mockInput,
+            resource: resourceOperation,
+            config: { mock: { dynamic: false } },
+          });
+
+          describe('the property has an example key', () => {
+            it('should return the example key', () => expect(response.body).toHaveProperty('name', 'Clark'));
+            describe('and a default key', () => {
+              it('prefers the example', () => expect(response.body).toHaveProperty('middlename', 'J'));
+            });
+
+            describe('and the property containing the example is deeply nested', () => {
+              it('should return the example key', () => expect(response.body).toHaveProperty('pet.name', 'Clark'));
+              it('should still prefer the example', () => expect(response.body).toHaveProperty('pet.middlename', 'J'));
+            });
+          });
+
+          describe('the property has not an example, but a default key', () => {
+            it('should use such key', () => {
+              expect(response.body).toHaveProperty('surname', 'Kent');
+            });
+          });
+
+          describe('the property has nor default, nor example', () => {
+            describe('is nullable', () => {
+              it('should be set to null', () => expect(response.body).toHaveProperty('age', null));
+            });
+
+            describe('and is not nullable', () => {
+              it('should return the default string', () => expect(response.body).toHaveProperty('email', 'string'));
+              it('should return the default number', () => expect(response.body).toHaveProperty('deposit', 0));
+              it('should return the first enum value', () =>
+                expect(response.body).toHaveProperty('paymentStatus', 'completed'));
+              it('should return the first anyOf value', () =>
+                expect(response.body).toHaveProperty('creditScore', 1958));
+              it('should return the first oneOf value', () =>
+                expect(response.body).toHaveProperty('paymentScore', 'string'));
+              it('should return the first allOf value', () =>
+                expect(response.body).toHaveProperty('walletScore', 'hello'));
+            });
+          });
+        });
```

Comment:
Note to self: do we actually need to test any of this? If I'm not mistaken we are using the library to generate this content which means we're effectively testing a library here... If so we should probably just mock the library and see it got called (?)

Reply:
Good question. The thing is that the horrible function you just saw will be removed when openapi-sampler's PR will be merged and released (hopefully very soon).

This test will effectively make sure that the examples generated by the updated library still match with what we want. Does it clarify?